File: app/src/main/scala/tripaint/TriPaintModel.scala
package tripaint

import tripaint.grid.ImageGrid
import tripaint.image.ImagePool
import tripaint.infrastructure.FileSystem

class TriPaintModel(val fileSystem: FileSystem, initialImageSize: Int) {
  val imagePool: ImagePool = new ImagePool()
  val imageGrid: ImageGrid = new ImageGrid(initialImageSize)
}

object TriPaintModel {
  def create(): TriPaintModel = new TriPaintModel(FileSystem.create(), -1)

  def createNull(
      imageSize: Int,
      fileSystemArgs: FileSystem.NullArgs = FileSystem.NullArgs()
  ): TriPaintModel =
    new TriPaintModel(FileSystem.createNull(fileSystemArgs), imageSize)
}

File: app/src/main/kotlin/tripaint/app/TriPaintModel.kt
package tripaint.app

import tripaint.grid.ImageGrid
import tripaint.image.ImagePool

class TriPaintModel(val fileSystem: FileSystem, initialImageSize: Int) {
    val imagePool: ImagePool = ImagePool()
    val imageGrid: ImageGrid = ImageGrid(initialImageSize)

    companion object {
        fun create(): TriPaintModel = TriPaintModel(FileSystem.create(), -1)

        fun createNull(
            imageSize: Int,
            fileSystemArgs: FileSystem.NullArgs = FileSystem.NullArgs()
        ): TriPaintModel = TriPaintModel(FileSystem.createNull(fileSystemArgs), imageSize)
    }
}
